app/models/client.rb

```diff
@@ -7,14 +7,32 @@ def full_name
   end
 
   def contacted_at
-    # the date of the most recent message sent or received
+    # the date of the most recent message sent to or received from this client
     last_message = Message.where(client: self).order(:created_at).last
     if last_message
       return last_message.created_at
     end
     self.updated_at
   end
 
+  def received_message_at
+    # the date of the most recent message received from this client
+    last_received_message = Message
+      .where(client: self, inbound: true)
+      .order(:created_at)
+      .last
+
+    if last_received_message
+      return last_received_message.created_at
+    end
+    self.updated_at
```

Comment:
I'm not sure exactly what interface that the callers of this method are expecting(*), but I would be somewhat surprised if the client has never received a message yet this method returns the `updated_at` timestamp. It seems like making this method return `nil` if the client has never received a message might be less surprising.

\* Am I reading this right that there is nothing calling this method yet?